### PR TITLE
Fix parsing issue with MLS with embedded strings

### PR DESCRIPTION
**Pull request by FloLey**
Currently the parser fails to parse MLSes correctly if there is a string enclosed at the end of it.
(when we have 4 consecutive `"` like in `exec::in_shell("""grep "test"""")`
This commit fixes the regex to accept strings (opened and closed with `"`) in a MLS

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Review by wouterdb — APPROVED** [review]
My regex fu is not sufficiently strong here.

**Comment by FloLey** [comment]
If you want to play with the regex, I build it here: [https://regex101.com/r/J8kcdf/1](https://regex101.com/r/J8kcdf/1)

**Review by sanderr — APPROVED** [review]

**Review by sanderr — CHANGES_REQUESTED** [review]
I meant to request changes, oops.

**Comment by sanderr** [comment]
Issue with original regex confirmed and suggestion seems to work.

**Comment by sanderr** [comment]
(https://regex101.com/r/dXe4Jc/1)

File: src/inmanta/parser/plyInmantaLex.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def t_JCOMMENT(t: lex.LexToken) -> None:  # noqa: N802
 
 
 def t_MLS(t: lex.LexToken) -> lex.LexToken:
-    r'"{3}([\s\S]*?)"{3}'
+    r'"{3}(((\"([^"]+)*\")|[\s\S])*?)"{3}'
```
[the remark this answers is not in the document]
Comment (FloLey) [reply src/inmanta/parser/plyInmantaLex.py:134]: So stupid of me, I looked so hard to handle "string" when I should just accept " at the end...

But why 5 and not 4?

[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/parser/plyInmantaLex.py:134]: I think two strings should be valid two, no? `"""echo Here, a string: """""`.

[the remark this answers is not in the document]
Comment (FloLey) [reply src/inmanta/parser/plyInmantaLex.py:134]: indeed, thanks :p

[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/parser/plyInmantaLex.py]: My proposal does allow for `""""""""""` to represent the mls containing 4 quotes though, that's not entirely correct  but I'm not sure how far we want to go to be robust against every edge case. @wouterdb?

File: tests/test_parser.py [hunk 2]
```diff
@@ -1403,6 +1403,57 @@ def test_mls_7():
...
+    )
+    assert len(statements) == 1
+    mls = statements[0]
+
+    assert isinstance(mls, LocatableString)
+    assert mls.lnr == 2
+    assert mls.elnr == 2
+    assert mls.start == 1
+    assert mls.end == 17
+    assert str(mls.value) == 'String: ""'
```
Comment (sanderr) [inline tests/test_parser.py:1420]: It's interesting to see how GitHub seems to have issues with identifying mls strings for syntax highlighting as well. Guess we shouldn't feel to bad then.

**Review by sanderr — APPROVED** [review]

File: src/inmanta/parser/plyInmantaLex.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanderr) [reply src/inmanta/parser/plyInmantaLex.py]: Rerequesting @wouterdb 's review for confirmation.

**Review by wouterdb — APPROVED** [review]
